### PR TITLE
App Properties, key management.

### PR DESCRIPTION
**Pull request by avilanicolas**
You'll have to add your own credentials.key
file for this to work. The key must be a valid
AES encrypted key. (128 bits)

Addresses:
  #2 
  #27

### CODE REVIEW

File: server/application.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# List of all usernames authorized to use this service
+remoteTinyFS.registeredUsers=avilan,jarhar,foaad
```
Comment (josepharhar) [inline server/application.properties]: I still don't really agree with this but if you insist

File: server/src/main/java/com/tinyfs/config/CredentialsConfig.java [hunk 2]
```diff
@@ -1,27 +1,50 @@
...
+      AES_KEY_LENGTH);
+    return new SecretKeySpec(
+      bytes,
+      0,
+      AES_KEY_LENGTH,
+      CREDENTIALS_ENCRYPTION_SCHEME);
+  }
+
+  private String getKeyText(final String resourceName) throws CredentialsException {
+    File file = new File(CredentialsConfig.class.getClassLoader().getResource("credentials.key").getFile());
```
Comment (josepharhar) [inline server/src/main/java/com/tinyfs/config/CredentialsConfig.java]: As I mentioned in issue #2 I think it would be best if we could randomly generate a private key if there isnt one already found. What is even the process for making a private key? I am getting a really big stacktrace when I start the server, and I assume its because I haven't made a credentials.key file

Comment (avilanicolas) [inline server/src/main/java/com/tinyfs/config/CredentialsConfig.java]: While I could be convinced to do this, I am not really convinced it's a good idea. I'd rather have the keys work across server instances / builds. I've sent you the key privately but I'll resend it to you.

It also should fail without a crednetials.key file...

Comment (avilanicolas) [inline server/src/main/java/com/tinyfs/config/CredentialsConfig.java]: I'd be fine making the server generate it if it doesn't have one, but I'd rather do it in a separate PR.
